File: src/main/java/com/github/dwendelen/platformd/infrastructure/authentication/IdentityProvider.java
package com.github.dwendelen.platformd.infrastructure.authentication;

import com.github.dwendelen.platformd.core.user.User;
import org.springframework.stereotype.Component;

@Component
public class IdentityProvider {
    private ThreadLocal<User> currentUser = new ThreadLocal<>();

    public User getCurrentUser() {
        User user = currentUser.get();
        if(user == null) {
            throw new UnsupportedOperationException("Decent exception");
        }
        return user;
    }

    public void setCurrentUser(User user) {
        currentUser.set(user);
    }

    public void clearCurrentUser() {
        currentUser.remove();
    }
}

File: application/src/main/scala/com/github/dwendelen/platformd/infrastructure/authentication/IdentityProvider.scala
package com.github.dwendelen.platformd.infrastructure.authentication

import com.github.dwendelen.platformd.rest.domain.user.User
import org.springframework.stereotype.Component

@Component
class IdentityProvider {
  private val currentUser = new ThreadLocal[User]

  def getCurrentUser: User = {
    val user = currentUser.get
    if (user == null) {
      throw new UnsupportedOperationException("Decent exception")
    }
    user
  }

  def setCurrentUser(user: User) {
    currentUser.set(user)
  }

  def clearCurrentUser() {
    currentUser.remove()
  }
}
